Feature/185 upgrade bootstrap to 4 (#83)
chameleon-system/chameleon-base: UPGRADE-6.3.md

## Database Fields

None.

> ## Backend Theme Library

Reviewer: We should a notice for the changed jQuery version, too (even if the version might change again until release).

Author: This PR ist totally immediate. I already added this info in the follow up PR.

> The Backend was upgraded to Bootstrap 4.1.3

Reviewer: ```suggestion
The Backend was upgraded to Bootstrap 4.1.3.
```

Author: fixed

See the [Bootstrap Migration Guide](https://getbootstrap.com/docs/4.1/migration/) for needed changes to your backend modules.

During the upgrade to Bootstrap 4 the following styles where checked and these are our findings:

> not found anywhere (so you may skip this search, too):

Reviewer: ```suggestion
Not found anywhere (so you might want to skip this search, too):
```

Author: fixed.

> not found anywhere (so you may skip this search, too):

Reviewer: Could we move this list to the end of the section? This would make more clear where the list ends, as the other lists don't have an introduction text.

Author: you are right. Done.

> - Some TCMSField types and TCMSTableEditors so check yours) 

Reviewer 2: ) too much

What does this line mean? There is html code in those classes?

Author: Yes. It is. In most of older/all fieldtypes there is HTML code right in the class.

- thumbnail
- list-line
- page-header
- dl-horizontal
- blockquote
- btn-xs
- btn-group-justified
- btn-group-xs
- breadcrumb
- center-block
- img-responsive
- img-rounded
- form-horizontal
- radio
- checkbox
- input-lg
- control-label 
- hidden/visible-xs, sm, md, lg
- label
- navbar-form
- navbar-btn
- progress-bar*

btn-default -> btn-secondary

> - some TCMSField types and TCMSTableEditors so check yours) 

Reviewer: ```suggestion
- Some TCMSField types and TCMSTableEditors so check yours) 
```

Author: fixed.

> .pull-left > .float-left

Reviewer: ```suggestion
.pull-left -> .float-left
```

Author: changed everywhere in the readme

- TCMSFieldDocument
- TCMSFieldDocumentProperties
- TCMSFieldMediaProperties
- TCMSFieldModuleInstance
- MTHeader
- header navigation
- footer

> - added CSS class migration in TCMSRender::DrawButton method for backwards compatibility

Reviewer: ```suggestion
- Added CSS class migration in TCMSRender::DrawButton method for backwards compatibility
```

Author: fixed.

> .input-sm > .form-control-sm

Reviewer: ```suggestion
.input-sm -> .form-control-sm
```

Author: changed

> - almost all TCMSField classes and twig templates

Reviewer: ```suggestion
- Almost all TCMSField classes and Twig templates
```

Author: fixed.

> - some list managers

Reviewer: ```suggestion
- Some list managers
```

Author: fixes.

> - some TCMSTableEditor classes

Reviewer: ```suggestion
- Some TCMSTableEditor classes
```

Author: fixed.

.table-condensed > .table-sm
- TCMSFieldDocumentProperties
- TCMSFieldMediaProperties
- TFullGroupTable

New: .page-item + .page-link

> - pagination in TFullGroupTable

Reviewer: ```suggestion
- Pagination in TFullGroupTable
```

Author: fixed.

.pull-rigt > .float-right
- TFullGroupTable
- MTHeader
- MTTableditor

.input-group-addon > .input-group-append

> - field types using the text length counter addon (varchar)

Reviewer: ```suggestion
- Field types using the text length counter addon (varchar)
```

Author: fixed.
